File: internal/model/folder.go
package model

import "time"

// DiskFolder 目录结构表
type DiskFolder struct {
	ID         uint64    `gorm:"primaryKey;autoIncrement" json:"id"`
	UserID     string    `gorm:"index;size:64;not null" json:"userId"`
	ParentID   uint64    `gorm:"index;default:0" json:"parentId"`
	FolderName string    `gorm:"size:255;not null" json:"folderName"`
	FullPath   string    `gorm:"size:1024;not null" json:"fullPath"`
	SortOrder  int       `gorm:"default:0" json:"sortOrder"`
	IsDeleted  bool      `gorm:"default:false" json:"isDeleted"`
	CreatedAt  time.Time `gorm:"autoCreateTime" json:"createdAt"`
	UpdatedAt  time.Time `gorm:"autoUpdateTime" json:"updatedAt"`
}

// TableName handles the TableName endpoint.
func (DiskFolder) TableName() string { return "disk_folder" }

File: sdk/src/agentdisk/models/folder.py
"""Folder models."""

from dataclasses import dataclass


@dataclass
class DiskFolder:
    id: int
    user_id: str
    parent_id: int
    folder_name: str
    full_path: str
    sort_order: int
    is_deleted: bool
    created_at: str
    updated_at: str

    @classmethod
    def from_dict(cls, d: dict) -> "DiskFolder":
        return cls(
            id=d["id"],
            user_id=d.get("userId", ""),
            parent_id=d.get("parentId", 0),
            folder_name=d.get("folderName", ""),
            full_path=d.get("fullPath", ""),
            sort_order=d.get("sortOrder", 0),
            is_deleted=d.get("isDeleted", False),
            created_at=d.get("createdAt", ""),
            updated_at=d.get("updatedAt", ""),
        )
